cmd/create.go

```diff
@@ -385,11 +399,31 @@ func generatePublicKey(key *rsa.PublicKey) ([]byte, error) {
 	return pubKeyBytes, nil
 }
 
-func encodePrivateKeyToPEM(key *rsa.PrivateKey) []byte {
-	privDER := x509.MarshalPKCS1PrivateKey(key)
+func encodePrivateKeyToPEM(key any, blockType string) []byte {
+	blockTypeDefault := "PRIVATE KEY"
+
+	if len(blockType) > 0 {
+		blockTypeDefault = " " + blockTypeDefault
+	}
+	blockType = blockType + blockTypeDefault
+
+	var privDER []byte
+	var err error
+	if strings.Contains(blockType, "OPENSSH") {
+		// Necessary to cast type in order to correctly marshal the key for OpenSSH
+		privDER, _ = x509.MarshalPKCS8PrivateKey(key.(ed25519.PrivateKey))
+	} else if strings.Contains(blockType, "RSA") {
+		privDER = x509.MarshalPKCS1PrivateKey(key.(*rsa.PrivateKey))
+	} else {
+		privDER, err = x509.MarshalPKCS8PrivateKey(key)
+	}
+
+	if err != nil {
+		logrus.Fatal("failed to marshal PKCS8 private key: ", err)
+	}
```

Comment:
```suggestion
	var privBytes []byte
	var err error
	if strings.Contains(blockType, "OPENSSH") {
		// Necessary to cast type in order to correctly marshal the key for OpenSSH
		privBytes, err = x509.MarshalPKCS8PrivateKey(key.(ed25519.PrivateKey))
		if err != nil {
			logrus.Fatalf("failed to marshal PKCS8 private key: %v", err)
		}
	} else if strings.Contains(blockType, "RSA") {
		privBytes = x509.MarshalPKCS1PrivateKey(key.(*rsa.PrivateKey))
	} else { // is this block expected? or should it fatal
		privBytes, err = x509.MarshalPKCS8PrivateKey(key)
		if err != nil {
			logrus.Fatalf("failed to marshal PKCS8 private key: %v", err)
		}
	}
```

Reply:
I believe we should leave it in, in case someone wants to use some other key type? Otherwise I can rip this out and just fatal.